src/core/overlay.ts

```diff
@@ -24,64 +23,107 @@ export class Overlay {
   public run(spec: APIDefinition, overlay: OpenAPIOverlay): APIDefinition {
     // Use jsonpath.apply to do the changes
-    if (overlay.actions && overlay.actions.length > 0)
+    if (overlay.actions && overlay.actions.length > 0) {
       for (const a of overlay.actions) {
         const action = a as JSONSchema4Object
         if (!action.target) {
-          process.stderr.write('Action with a missing target\n')
+          process.stderr.write(`WARNING: ${this.humanName(action)} has an empty target\n`)
           continue
         }
 
         const target = action.target as string
-        // Is it a remove?
-        if (Object.hasOwn(action, 'remove')) {
-          /* eslint-disable-next-line no-constant-condition */
-          while (true) {
-            const path = jsonpath.paths(spec, target)
-            if (path.length === 0) {
-              break
-            }
-
-            const parent = jsonpath.parent(spec, target)
-            const thingToRemove = path[0].at(-1)
-            if (thingToRemove !== undefined) {
-              if (Array.isArray(parent)) {
-                parent.splice(thingToRemove as number, 1)
-              } else {
-                delete parent[thingToRemove]
-              }
-            }
-          }
-        } else {
-          try {
-            // It must be an update
-            // Deep merge objects using a module (built-in spread operator is only shallow)
-            const merger = mergician({appendArrays: true})
-            if (target === '$') {
-              // You can't actually merge an update on a root object
-              // target with the jsonpath lib, this is just us merging
-              // the given update with the whole spec.
-              spec = merger(spec, action.update)
-            } else {
-              jsonpath.apply(spec, target, (chunk) => {
-                if (typeof chunk === 'object' && typeof action.update === 'object') {
-                  if (Array.isArray(chunk) && Array.isArray(action.update)) {
-                    return [...chunk, ...action.update]
-                  }
-
-                  return merger(chunk, action.update)
-                }
+        // jsonpathly's paths are strings. They represent a “Path
+        // expression” which represents the absolute path in the objet
+        // tree reached by our `target`.
+        //
+        // E.g. '$["store"]["book"][0]["price"]'
+        const paths = jsonpath.paths(spec, target)
+        if (paths.length === 0) {
+          process.stderr.write(`WARNING: Action target '${target}' has no matching elements\n`)
+          continue
+        }
 
-                return action.update
-              })
-            }
-          } catch (error) {
-            process.stderr.write(`Error applying overlay: ${(error as Error).message}\n`)
-            // return chunk
-          }
+        for (const path of paths) {
+          this.executeAction(spec, action, path)
```

Comment:
During refacto, I understood that `spec` is modified each time this function `executeAction` is called.
May I suggest to add a comment here to highlight this?